cpp/oneapi/dal/table/backend/homogen_table_builder_impl.hpp

```diff
@@ -35,8 +37,8 @@ class homogen_table_builder_impl {
         dtype_ = data_type::float32;
     }
 
-    void reset(homogen_table&& t) {
-        if (t.has_data()) {
+    void move(detail::homogen_table_iface& t) override {
```

Comment:
It is strange to me that `move` accepts lvalue ref

Reply:
Hmm, `std::move` accepts lvalue (but not limited by lvalue)

Reply:
But the goal of `std::move` is different. I'm trying to say this name confuses me because it actually does not move the data of `t` inside the builder. It refers to the data of `t` if it is not empty. I'd expect that `move()` method will make `t` empty. Probably, `reset` of `refer` is a better name. What were the reasons to change the signature of that function?